### ISSUE TITLE
Will this support listener to specific event?

### ISSUE THREAD

**Issue by flyinghawker** [dart-lang/web_socket_channel#69]
Specific event listener is necessary in some scenarios. Hope to have it.

**Comment by natebosch** [comment]
What do you mean by "event"? A web socket is a duplex communication channel - it doesn't define anything in terms of "event".

The `Stream` API  doc talk about "events" - in as much as you can get the stream for a web socket channel you can deal with events in any way you can with a stream, but I don't know what "specific" would mean in that context.

Perhaps you're meaning events like DOM events? In that case you'll want to look at the APIs around [`dart:html`](https://api.dart.dev/stable/2.4.0/dart-html/dart-html-library.html)

**Comment by tgarm** [comment]
I guess 'events' here might be 'exceptions'. Such as:

- connected

- disconnected

- message sent

-  message send failed

**Comment by jsu93** [comment]
connected is the `onOpen` event, but I didn't see any api about this in this package.

https://api.dartlang.org/stable/2.4.0/dart-html/WebSocket/onOpen.html

**Comment by natebosch** [comment]
I don't think this package is not intended to expose the low level plumbing - for that use the `WebSocket` directly. This package is wrapping an existing `WebSocket` and exposing it as a `StreamChannel`.

We do expose at least a little beyond the `StreamChannel` interface, with things like `closeCode`, but I'm not sure how much we'd want to expand that...

**Comment by cTatu** [comment]
> Specific event listener is necessary in some scenarios. Hope to have it.

I made an implementation **[here](https://pub.dev/packages/websocket_channel_wrapper)** of [this npm wrapper](https://github.com/bminer/ws-wrapper) over the IOWebSocketChannel. Implements requests and messages based on events like [socket.io](https://github.com/infitio/flutter_socket_io)
